scripts/summarize-changes.js

```diff
@@ -0,0 +1,363 @@
+/**
+ * This script is used to summarize the changes to the interface definitions in the
+ * `src/internal/reference/generated` directory between the current HEAD and the
+ * previous commit.
+ *
+ * It should be run from the root of the repo immediately after rebuilding the
+ * generated files - this is handled by the `build:clean` script already.
+ *
+ * It will output a markdown file to `docs/changelog` with the following format:
+ *
+ * # Interface Change Summary: 1.0.0
+ *
+ * ## src/internal/reference/generated/channels.v3.ts
+ *
+ * - **Added Interface:** \`ChannelNode\`
+ * - **Removed Interface:** \`ChannelNode\`
+ *
+ */
+// TECH DEBT: This is a naive approach that uses regex to remove JSDocs and find parents properties.
+// It can probably be improved by leaning more on ts-morph to parse the AST.
+import { execSync } from 'child_process';
+import { readFileSync, writeFileSync, existsSync, mkdirSync } from 'fs';
+import { resolve } from 'path';
+import { Project, SyntaxKind } from 'ts-morph';
+import * as Diff from 'diff';
+
+const packageJson = JSON.parse(readFileSync('package.json', 'utf-8'));
+
+const config = {
+    version: packageJson.version,
+    baseRef: 'HEAD~1',
+    targetDir: 'src/internal/reference/generated',
+    outputDir: `docs/changelog`
+}
+
+function formatOutputFileName(config) {
+    return `${config.outputDir}/${config.version}.md`;
+}
+
+/**
+ * Gets the content of a file at a specific Git reference.
+ * @param {string} gitRef
+ * @param {string} filePath - Path relative to repo root
+ * @returns {string} The file content or an empty string if the file doesn't exist at that ref.
+ */
+function getFileContentAtRef(gitRef, filePath) {
+    try {
+        const contentBuffer = execSync(`git show ${gitRef}:${filePath}`, { encoding: 'buffer', stdio: 'pipe' });
+        return contentBuffer.toString('utf-8');
+    } catch (error) {
+        // Handle case where file didn't exist at that ref (e.g., new file)
+        // Check stderr for common "does not exist" messages
+        const stderr = error.stderr ? error.stderr.toString() : '';
+        if (stderr.includes('exists on disk, but not in') || stderr.includes('does not exist')) {
+            return '';
+        }
+        console.warn(`Warning: Could not get content of ${filePath} at ${gitRef}. Error: ${error} Stderr: ${stderr}`);
+        return '';
```

Comment:
I think we can utilise something like [simple-git](https://github.com/steveukx/git-js) for this part in the future.